catalog/files/api/src/main/java/org/projectnessie/catalog/files/api/BackendErrorStatus.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.catalog.files.api;
+
+import static org.projectnessie.catalog.files.api.BackendErrorCode.FORBIDDEN;
+import static org.projectnessie.catalog.files.api.BackendErrorCode.NOT_FOUND;
+import static org.projectnessie.catalog.files.api.BackendErrorCode.UNAUTHORIZED;
+import static org.projectnessie.catalog.files.api.BackendErrorCode.UNKNOWN;
+
+import org.projectnessie.nessie.immutables.NessieImmutable;
+
+@NessieImmutable
+public interface BackendErrorStatus {
+  BackendErrorCode statusCode();
+
+  Throwable cause();
+
+  static BackendErrorStatus of(BackendErrorCode statusCode, Throwable cause) {
+    return ImmutableBackendErrorStatus.of(statusCode, cause);
+  }
+
+  static BackendErrorStatus fromHttpStatusCode(int httpStatusCode, Throwable cause) {
+    switch (httpStatusCode) {
```

Comment:
Do we need 429 here as well?

Reply:
I can add it, but currently it'll probably be a dead branch. I'd like to use the storage client library's retry capabilities when we can (follow-up PR).

Reply:
Actually, I added it for the sake of completeness.